ichnaea/locate/provider.py

```diff
@@ -596,6 +600,17 @@ def should_locate(self, data, location):
     def locate(self, data):
         location = self.location_type(query_data=False)
 
+        if self.redis_client:
+            rate_key = 'fallback:{time}'.format(time=int(time.time()))
+            limit_reached = rate_limit(
+                self.redis_client,
+                rate_key,
+                maxreq=self.settings['ratelimit'],
+                expire=1,
```

Comment:
I'd set this to a bit more than 1 second, maybe 60 seconds. Since the timestamp is client side generated, it's susceptible to clock-skew (dean tells me we are usually seeing a couple seconds, outliers at 10-20 seconds of skew on machines). I wouldn't worry about that too much for this use-case, but keeping the keys around for a bit longer shouldn't hurt. A client that is a couple second behind the others, would still try to increment one of the existing older keys, and get throttled by that. If the keys are gone, we could have each of the three servers essentially write and increment its own counter without ever overlapping with the other counters.